clikit/src/main/java/org/adamalang/clikit/model/Argument.java

```diff
@@ -0,0 +1,73 @@
+package org.adamalang.clikit.model;
+
+import org.adamalang.clikit.exceptions.XMLFormatException;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class Argument {
+    public boolean optional;
+    public String name;
+    public String shortName;
+    public String defaultValue = "";
+    public ArgDefinition definition;
+    public String camel;
+
+    public Argument(String name, boolean optional, String defaultValue, ArgDefinition definition) {
+        this.optional = optional;
+        this.name = name;
+        this.camel = Common.camelize(name, true);
+        this.defaultValue = defaultValue;
+        this.definition = definition;
+    }
+    public Argument(String name, boolean optional) {
+        this.optional = optional;
+        this.camel = Common.camelize(name, true);
+        this.name = name;
+    }
+
+
+    public Argument(String name) {
+        this.optional = false;
+        this.camel = Common.camelize(name, true);
+        this.name = name;
+    }
```

Comment:
so, this is "fine"; however, I've come to conclude that having exactly one constructor is ideal. For example, these constructors allow null to come into play. As a design goal, I prefer all classes to me immutable by having every field final ( see above). When you make every field final, errors in various constructor will become clear.